File: RenResourceService/src/main/java/org/sysu/renResourcing/resource/Human.java
package org.sysu.renResourcing.resource;

import java.util.ArrayList;

/**
 * Author: Rinkako
 * Date  : 2017/12/15
 * Usage : Human Resource Data Package
 */
public class Human extends RResource {
    /**
     * Construct a new Human Entity.
     */
    public Human() {
        this.Role = new ArrayList<>();
    }

    /**
     * Add a role to this human.
     * @param roleId role id
     * @return did operation success
     */
    public boolean AddRole(String roleId) {
        if (this.Role.contains(roleId)) {
            return false;
        }
        this.Role.add(roleId);
        return true;
    }

    /**
     * Remove a role from this human.
     * @param roleId role id
     * @return did operation success
     */
    public boolean RemoveRole(String roleId) {
        if (this.Role.contains(roleId)) {
            this.Role.remove(roleId);
            return true;
        }
        return false;
    }

    /**
     * human person id, unique.
     */
    public String PersonID = "";

    /**
     * human first name, not unique.
     */
    public String FirstName = "";

    /**
     * human last name, not unique.
     */
    public String LastName = "";

    /**
     * A list of role which this human has.
     */
    protected ArrayList<String> Role = null;
}

File: COrgan/Entity/Human.py
#!/usr/bin/env python
# encoding: utf-8
"""
@module : Human
@author : Rinkako
@time   : 2017/12/26
"""
from Entity.Worker import Worker


class Human(Worker):
    """
    entity class for `human` data package.
    """

    def __init__(self, humanId, fName, lName, note):
        """
        Create new human instance.
        :param humanId: unique human id
        :param fName: first name
        :param lName: last name
        :param note: description of this human
        """
        Worker.__init__(self, "", note)
        self.PersonId = humanId
        self.FirstName = fName
        self.LastName = lName

    def ToJsonDict(self):
        """
        Get json dict.
        :return:
        """
        return {
            'GlobalId': self.GlobalId,
            'Note': self.Note,
            'PersonId': self.PersonId,
            'FirstName': self.FirstName,
            'LastName': self.LastName
        }
